src/store/flow/flow.ts

```diff
@@ -572,20 +575,21 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
   },
 
   async flow_removeAllSelectedBlocks({state, dispatch}) {
-    forEach(state.selectedBlocks, (blockId: IBlock['uuid']) => {
-      dispatch('flow_removeBlock', {blockId})
-    })
+    const removeBlock_actions = state.selectedBlocks.map((blockId: IBlock['uuid']) =>
+      dispatch('flow_removeBlock', {blockId}))
+
+    await Promise.all(removeBlock_actions)
 
     state.selectedBlocks = []
   },
```

Comment:
This kind of `Promise.all` logic (here and elsewhere) is needed to make sure that everything is done before we save the snapshot